src/algebraic_geometry/prime_spectrum/basic.lean

```diff
@@ -37,79 +37,87 @@ whereas we denote subsets of prime spectra with `t`, `t'`, etc...
 
 ## Inspiration/contributors
 
-The contents of this file draw inspiration from
-<https://github.com/ramonfmir/lean-scheme>
+The contents of this file draw inspiration from <https://github.com/ramonfmir/lean-scheme>
 which has contributions from Ramon Fernandez Mir, Kevin Buzzard, Kenny Lau,
 and Chris Hughes (on an earlier repository).
-
 -/
 
 noncomputable theory
 open_locale classical
 
 universes u v
 
-variables (R : Type u) [comm_ring R]
+variables (R : Type u) (S : Type v) [comm_ring R] [comm_ring S]
 
-/-- The prime spectrum of a commutative ring `R`
-is the type of all prime ideals of `R`.
+/-- The prime spectrum of a commutative ring `R` is the type of all prime ideals of `R`.
 
 It is naturally endowed with a topology (the Zariski topology),
 and a sheaf of commutative rings (see `algebraic_geometry.structure_sheaf`).
 It is a fundamental building block in algebraic geometry. -/
-@[nolint has_nonempty_instance]
-def prime_spectrum := {I : ideal R // I.is_prime}
+@[ext] structure prime_spectrum :=
+(as_ideal : ideal R)
```

Comment:
This should probably be called `to_ideal`, but that probably makes sense as a separate PR

Reply:
There is a (short discussion)[https://github.com/leanprover-community/mathlib/pull/16905] about this, where we want `as_ideal` instead of `to_ideal` to reflect that we're thinking of the ideals in the prime spectrum as points rather than as ideals with an extra field.